### ISSUE TITLE
error in reorg layer?

### ISSUE THREAD

**Issue by fasterbuild** [AlexeyAB/darknet#203]
Hi AlexeyAB and everyone,
    Below is the reorg code and i'm confused about it.

void reorg_cpu(float *x, int w, int h, int c, int batch, int stride, int forward, float *out)
{
    int b,i,j,k;
    int out_c = c/(stride*stride);

    for(b = 0; b < batch; ++b){
        for(k = 0; k < c; ++k){
            for(j = 0; j < h; ++j){
                for(i = 0; i < w; ++i){
                    int in_index  = i + w*(j + h*(k + c*b));
                    int c2 = k % out_c;
                    int offset = k / out_c;
                    int w2 = i*stride + offset % stride;
                    int h2 = j*stride + offset / stride;
                    int out_index = w2 + w*stride*(h2 + h*stride*(c2 + out_c*b));
                    if(forward) out[out_index] = x[in_index];
                    else out[in_index] = x[out_index];
                }
            }
        }
    }
}

The w,h,c are input params. When inference, they are 26*26*512 in 416*416 image, and the output is expected to be 13*13*2048。Then the out_c doesn't  make sense. I print some in_index and out_index in pair and doesn't find any mapping relationship between them.
I think the w,h,c should be output params, then all go right.
And the variable "forward" is misleading because when inference the "forward" is 0 and code goes to "out[in_index] = x[out_index]".
AlexeyAB would please Check them?

**Comment by fasterbuild** [comment]
I'm sorry I have mistaken the meaning of reorg layer. AlexeyAB is right.

**Comment by AlexeyAB** [comment]
I agree that this is a very confusing repositioning of the elements, but it works :)

Permute 26x26x64 -> 13x13x256
But `c=64` and `out_c = c/(stride * stride)=16` - but where is 256? :)))
(for stride=2)

So in `out[in_index] = x[out_index];`

* out_index - is really input index, that uses `out_c=16`  and mistical `c2=0` - `15` (when really input channels=**64**) why? :)))

* in_index - is really output index, that uses `c=64` and `k=0` - `64` (when really output channels=**256**)

And `offset=0` - `3`, so in the input index (out_index) `w2=0` - `51` and `h2=0` - `51`, what are these numbers ? :)))

----

And for batch=1, i.e. b=0:

`int out_index = w2 + w*stride*(h2 + h*stride*c2);`=`[0-51]` + `26` * `2` * (`[0-51]` + `26` * `2` * `[0-15]`) = `[0 - 43263]`

Also `int in_index  = i + w*(j + h*(k + c*b));` too has max value is 43263

Ok, it should works.
